Feature(Widgets): add a search bar widget

mgermerie opened the pull request:
## Description
<!--- Describe your changes in detail -->
Add a search bar widget, allowing user to search locations through a specified geocoding service.

mgermerie committed:
feature(Widgets): add a searchbar widget

gchoqueux commented on examples/widgets_searchbar.html:
```diff
@@ -0,0 +1,115 @@
+                    'PositionOfInterest',
+                ),
+                parser: (result) => result.results.map(l => l.fulltext),
+                whenSelected: lookAtCoordinate,
```
could you replace `whenSelected` by `onSelected`

mgermerie replied:
Renamed.

gchoqueux commented on examples/widgets_searchbar.html:
```diff
@@ -0,0 +1,115 @@
+                    'https://wxs.ign.fr/ayxvok72rcocdyn8xyvy32og/ols/apis/completion?text=&type=StreetAddress,' +
+                    'PositionOfInterest',
+                ),
+                parser: (result) => result.results.map(l => l.fulltext),
```
could you add a comment to explain the parser?

could you write `response.results.` ?

could you map the coordinates?

mgermerie replied:
I did the changes

gchoqueux commented on utils/gui/Searchbar.js:
```diff
@@ -0,0 +1,252 @@
+
+    removeAllActives(htmlElements);
+    if (index >= htmlElements.length) { index = 0; }
+    if (index < 0) { index = (htmlElements.length - 1); }
```
`.at(-1)`?

mgermerie replied:
Cannot use `.at` method with `htmlElements` which is of type `HTMLCollection`

gchoqueux commented on utils/gui/Searchbar.js:
```diff
@@ -0,0 +1,252 @@
+            geocodingOptions.url.searchParams.set('text', value);
+
+            Fetcher.json(geocodingOptions.url).then((geocodingResult) => {
+                const resultArray = geocodingOptions.parser(geocodingResult);
```
could you rename `resultArray`?

mgermerie replied:
Done

gchoqueux commented on utils/gui/Searchbar.js:
```diff
@@ -0,0 +1,252 @@
+                        });
+
+                        autocompleteItem.addEventListener('click', () => {
+                            geocodingOptions.whenSelected(geocodingResult.results[i]);
```
could you use the map instead of `geocodingResult`

mgermerie replied:
Done

mgermerie commented:
I figured there is an issue with search bar and mouse interaction : when double-clicking the search bar on a `GlobeView`, you trigger a travel animation. On a `PlanarView` it appears the search bar is not clickable. This has to do with event propagation in `GlobeControls` and `PlanarControls`. I shall figure out a way to fix this.

mgermerie committed:
refacto(PlanarControls): change focus policy
- remove auto focus when hovering the view.domElement
- simplify implementation of focus on mouse click

mgermerie committed:
fix(Widgets): fix focus policy and event propagation on widgets
